embrace-gradle-plugin/src/main/java/io/embrace/android/gradle/plugin/tasks/ndk/CompressAndHashSharedObjectFilesTask.kt

```diff
@@ -0,0 +1,81 @@
+package io.embrace.android.gradle.plugin.tasks.ndk
+
+import io.embrace.android.gradle.plugin.hash.calculateSha1ForFile
+import io.embrace.android.gradle.plugin.tasks.EmbraceTaskImpl
+import io.embrace.android.gradle.plugin.util.compression.ZstdFileCompressor
+import io.embrace.android.gradle.plugin.util.serialization.MoshiSerializer
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import javax.inject.Inject
+
+/**
+ * Given a directory containing architecture directories (arm64-v8a, armeabi-v7a, etc.) with shared object files (.so files),
+ * this task:
+ * 1. Compresses each shared object file using Zstd compression
+ * 2. Stores the compressed files in compressedSharedObjectFilesDirectory, preserving the architecture directory structure
```

Comment:
I didn't realise we were doing Zstd compression on SO files. IMO it'd be worth breaking that into a separate task that takes the uncompressed SO files as an input and the compressed files as an output. Sorry for the confusion from our discussion yesterday